src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -1127,19 +1128,31 @@ def decrypt_key(cmd, client, algorithm, value, iv=None, tag=None, aad=None,
 
 
 def sign_key(cmd, client, algorithm, digest, name=None, version=None):
+    if '256' in algorithm:
+        hash_func = hashlib.sha256
+    elif '384' in algorithm:
+        hash_func = hashlib.sha384
+    else:
+        hash_func = hashlib.sha512
     SignatureAlgorithm = cmd.loader.get_sdk('SignatureAlgorithm', mod='crypto._enums',
                                             resource_type=ResourceType.DATA_KEYVAULT_KEYS)
     crypto_client = client.get_cryptography_client(name, key_version=version)
-    return crypto_client.sign(SignatureAlgorithm(algorithm), digest.encode('utf-8'))
+    return crypto_client.sign(SignatureAlgorithm(algorithm),
+                              hash_func(base64.b64decode(digest.encode('utf-8'))).digest())
```

Comment:
hash before signing as suggested in other language sample: https://learn.microsoft.com/en-us/azure/key-vault/keys/javascript-developer-guide-sign-verify-key#signing-data

Reply:
the argument is called digest, it's already a hash of the data. This is computing a hash of hash.